### DISCUSSION TITLE
Convert the kinematic orbits from CRF to along-track, cross-track and radial-track #2

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#137] [Q&A] [answered]
Hello

I have one additional question that is in the same area as discussion #121. If i have my orbit position results (incl. velocity and acceleration) in CRF then with SimulateStarCamera and InstrumentRotate i can move the "results" form CRF to along-track, cross-track, and radial-track directions. How would i transform the covariance matrix then to plot RMS values in the new reference frame? If I understand correctly with the input requirements for the 2 aforementioned scripts i cannot use those for a covariance matrix transformation? So, i would have to do it "outside of GROOPS" with a small script? Is my assumption correct or did I miss something somewhere? 

Txs for the help and clarification.

Cheers
Jean

**Comment by Bemtgenj** [comment]
Hello 

maybe i understood it now better - with SimulateStarCamera we create a unit vector and then we can add that output file together with the covariance file (same as the "orbit" file) into InstrumentRotate and get the covariance in the LOF frame. This means that we can transform any matrix like mentioned above.

TXS for the help any feedback

Cheers
Jean

**Comment by tmayerguerr** [comment, marked as answer]
Dear Jean,

SimulateStarCamera computes a rotation matrix from the satellite frame (along, cross, near radial axis) to the inertial frame (CRF) at each epoch. The rotation matrix is represented by a 4 element quaternion vector.
Different instrument data (positions, covariance matrices, ...) can be rotated to a new coordinate frame with InstrumentRotate.

regards
Torsten

**Comment by Bemtgenj** [comment]
Dear Torsten

Txs much for the feedback and the clarification

Cheers
Jean
